src/components/VDataTable/mixins/head.js

```diff
@@ -26,7 +26,7 @@ export default {
 
         children = [this.hasTag(row, 'th') ? this.genTR(row) : row, this.genTProgress()]
       } else {
-        const row = this.headers.map(o => this.genHeader(o))
+        const row = this.headers.map((o, i) => this.genHeader(o, this.headerKey ? o[this.headerKey] : i))
```

Comment:
Should we give a warning if `headerKey` is set but now defined in the object? Use the `consoleError` helper.

Reply:
Shall I add it?

Reply:
We don't add it here https://github.com/vuetifyjs/vuetify/blob/dev/src/components/VDataTable/mixins/body.js#L47, i don't think it's necessary